scripts/zkai_d128_two_slice_outer_proof_object_spike_gate.py

```
@@ -0,0 +1,1009 @@
+    except Exception as err:  # noqa: BLE001 - normalize imported validator errors.
+        raise D128TwoSliceOuterProofObjectSpikeError(f"failed to hash source evidence {path}: {err}") from err
+
+
+def relative_path(path: pathlib.Path) -> str:
+    return path.resolve().relative_to(ROOT.resolve()).as_posix()
+
+
+def _safe_candidate_path(raw_path: str) -> pathlib.Path:
+    pure_path = pathlib.PurePosixPath(require_str(raw_path, "candidate path"))
+    if pure_path.is_absolute() or raw_path != pure_path.as_posix() or any(part in ("", ".", "..") for part in pure_path.parts):
+        raise D128TwoSliceOuterProofObjectSpikeError(f"candidate path must be repo-relative: {raw_path}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)